docs/proposals/train_api_proposal.md

```diff
@@ -0,0 +1,137 @@
+**<h1>Train API Proposal for LLMs</h1>**
+
+**<h3>Authors:</h3>**
+
+* Deepanker Gupta (**[@deepanker13](https://github.com/deepanker13)**), Nutanix
+* Johnu George (**[@johnugeorge](https://github.com/johnugeorge)**), Nutanix
+
+**<h3>Status</h3>**
+
+* 10 Nov 2023 (v1)
+
+**<h3>Abstract</h3>**
+
+LLMs are being widely used for generative AI tasks and as their adoption is increasing across various domains, the need for LLMops in the Kubernetes environment has also risen. There is a need to fine-tune these large models on task-specific datasets. The user/data scientist should be able to do this simplistically in the Kubernetes environment using the Kubeflow training operator SDK without any infrastructure knowledge.
+
+**<h3>Background</h3>**
+
+Currently, there are two flows for data scientists to start using Training operator for their distributed training needs. 
+
+**<h4>Traditional method</h4>**
+
+1. The end-user has to write a custom script that includes data fetching, model loading, and a training loop using pytorch libraries.
+2. Create an image and push it to a registry.
+3. Using the Kubeflow train sdk to create worker spec and master spec which uses the above image and using these two specs create a pytorch job.
+
+This method requires infrastructure knowledge including image creation, registry access and training job spec. This makes it difficult for data scientists to purely focus at the application layer.
+
+**<h4>Higher level SDK</h4>**
+
+To provide a better user experience, a new higher level SDK was added in[ https://github.com/kubeflow/training-operator/pull/1659](https://github.com/kubeflow/training-operator/pull/1659)  which avoided the need of image creation. Instead of creating the image, we can directly pass the custom script to train_func parameter and final pytorch job is created
```

Comment:
```suggestion
To provide a better user experience, a new higher level SDK was added in[ https://github.com/kubeflow/training-operator/pull/1659](https://github.com/kubeflow/training-operator/pull/1659) which avoided the need of image creation. Instead of creating the image, we can directly pass the custom script to train_func parameter and final pytorch job is created
```